src/components/application_manager/include/application_manager/application.h

```diff
@@ -934,6 +935,52 @@ class Application : public virtual InitialApplicationData,
    */
   virtual const std::list<AppExtensionPtr>& Extensions() const = 0;
 
+  /**
+   * @brief Get map of pending button subscription requests correlation ids
+   * to button names
+   * @return pending button subscriptions map
+   */
+  const virtual std::map<int32_t, hmi_apis::Common_ButtonName::eType>&
+  PendingSubscriptionButtons() const = 0;
+
+  /**
+   * @brief Add  pending button subscription request
+   * @param correlation_id - correlation id of subscription request
+   * @param button_name - enum value indication button name
+   */
+  virtual void AddPendingSubscriptionButton(
+      const int32_t correlation_id,
+      const hmi_apis::Common_ButtonName::eType button_name) = 0;
+  /**
+   * @brief Remove pending button subscription request
+   * @param correlation_id - correlation id of subscription request
+   */
+  virtual void RemovePendingSubscriptionButton(
+      const int32_t correlation_id) = 0;
+
+  /**
+   * @brief Get map of pending button unsubscription requests correlation ids
+   * to button names
+   * @return pending button unsubscriptions map
+   */
+  const virtual std::map<int32_t, hmi_apis::Common_ButtonName::eType>&
```

Comment:
@mked-luxoft Move `const` qualifier to return type

Reply:
I think we need `const` here, because returning reference

Reply:
@AByzhynar fixed in 2b001c6014447a350d41a2bcd1e834fd2c83057a